plugins/JsTrackerInstallCheck/JsTrackerInstallCheck.php

```diff
@@ -0,0 +1,149 @@
+            return;
+        }
+
+        // If the nonce is older 30 seconds, ignore it. This should be plenty of time because an API call creates the nonce just before opening the site
+        if (empty($nonceOptionArray['time']) || Date::getNowTimestamp() - $nonceOptionArray['time'] > 30) {
+            return;
+        }
+
+        // Since the nonce matches and hasn't expired, update the option indicating success
+        $nonceOptionArray['isSuccessful'] = true;
+        Option::set(self::OPTION_NAME_PREFIX . $request->getIdSite(), json_encode($nonceOptionArray));
+    }
+
+    /**
+     * Check whether a test request has been recorded for the provided nonce. If no nonce is provided, the recorded
+     * result for the site will be returned.
+     *
+     * @param string $idSite
+     * @param string $nonce The unique nonce used to identify the test requests. Optionally can be left empty if simply
+     * wanting to check if the site has been successfully tested.
+     * @return array list of containers ['isSuccess' => true]
+     */
+    public function checkForJsTrackerInstallTestSuccess(string $idSite, string $nonce = ''): array
+    {
+        $nonceOptionString = Option::get(self::OPTION_NAME_PREFIX . $idSite);
+        if (empty($nonceOptionString)) {
+            return ['isSuccess' => false];
+        }
+
+        $nonceOptionArray = json_decode($nonceOptionString, true);
+        if (empty($nonceOptionArray)) {
+            return ['isSuccess' => false];
+        }
+
+        // Check if the nonce matches the recorded nonce
+        if (!empty($nonce) && (empty($nonceOptionArray['nonce']) || $nonceOptionArray['nonce'] !== $nonce)) {
+            return ['isSuccess' => false];
+        }
+
+        return ['isSuccess' => !empty($nonceOptionArray['isSuccessful'])];
+    }
+
+    /**
+     * Initiate a test whether the JS tracking code has been successfully installed for a site. It generates a nonce and
+     * stores it in the option table so that it can be accessed later during the Tracker.isExcludedVisit event.
+     *
+     * @param string $idSite
+     * @return array containing the URL constructed using the main URL for the site and the newly created nonce as a
+     * query parameter.
+     * E.g ['url' => 'https://some-site.com?tracker_install_check=c3dfa1abbbab6381baca0793b8dd5d', 'nonce' => 'c3dfa1abbbab6381baca0793b8dd5d']
+     * @throws \Exception
+     */
+    public function initiateJsTrackerInstallTest(string $idSite): array
+    {
+        $nonceString = md5(SettingsPiwik::getSalt() . time() . Common::generateUniqId());
+        Option::set(self::OPTION_NAME_PREFIX . $idSite, json_encode([
+            'nonce' => $nonceString,
+            'time' => Date::getNowTimestamp(),
+            'isSuccessful' => false
+        ]));
+
+        // Look up the site so that we can get the main URL
+        $site = SitesManagerApi::getInstance()->getSiteFromId($idSite);
+
+        $url = $site['main_url'];
```

Comment:
Would be simpler to use `Site::getMainUrlFor($idSite)`

Reply:
Yes. Thank you. I looked for a method like that, but didn't find one. I'll update to use that.